CBLClient/Apps/CBLTestServer-Java-Desktop/src/main/java/com/couchbase/mobiletestkit/javatestserver/TestServerMain.java

```diff
@@ -1,35 +1,158 @@
 package com.couchbase.mobiletestkit.javatestserver;
 
+import org.apache.commons.daemon.Daemon;
+import org.apache.commons.daemon.DaemonContext;
+import org.apache.commons.daemon.DaemonInitException;
+
 import com.couchbase.mobiletestkit.javacommon.Context;
 import com.couchbase.mobiletestkit.javalistener.Server;
 import com.couchbase.mobiletestkit.javacommon.util.Log;
 import com.couchbase.lite.CouchbaseLite;
 import org.nanohttpd.protocols.http.NanoHTTPD;
-import org.nanohttpd.util.ServerRunner;
 
 import java.io.File;
 import java.io.IOException;
-import java.io.InputStream;
 
-public class TestServerMain {
+
+public class TestServerMain implements Daemon {
     private static final int PORT = 8080;
     private static final String TMP_DIR = System.getProperty("java.io.tmpdir");
+    private static final String TAG = "TestServer-Java";
     private static Context context;
+    private static boolean stopped = false;
+    private static TestServerMain testserverLauncherInstance = null;
 
+    /**
+     * Main method runs as non-service mode for debugging use
+     * @param args
+     * @throws IOException
+     */
     public static void main(String[] args) throws IOException {
+        if(testserverLauncherInstance == null){
```

Comment:
Format:
- space between if and (
- space between) and {

Reply:
fixed